### PR TITLE
HADOOP-16202 Enhance S3A openFile()

### PR DESCRIPTION
**Pull request by steveloughran**
* s3a fs will use any FileStatus passed in to openFile...needed for mount/wrapped FS
* and supports an option "fs.s3a.open.option.length" to set the length -which will also trigger the HEAD being skipped
* builder spec updated to say status path is ignored
* add tracking of all option keynames passed to builder.

I had intended to support etag and version but its too tricky...you'd also need to supply the length or you issue HEAD with the given etag/version to extract that param too. The tracking of all option keys was for that. Not yet used -but left for others

needs new tests

-pass in status with different path (in contract)
-new option skips HEAD request (trivial test -open missing file, seek to below declared length, only expect failure on read())

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  20m 47s |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 33s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   2m 46s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  4s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  20m 26s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 28s |  trunk passed  |
| +0 :ok: |  spotbugs  |   1m  6s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  7s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 22s |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 53s |  the patch passed  |
| +1 :green_heart: |  javac  |  16m 53s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 49s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   2m  3s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m  7s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 30s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 11s |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 31s |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 45s |  The patch does not generate ASF License warnings.  |
|  |   | 122m 45s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2046 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle markdownlint |
| uname | Linux 2b0a1f09fe16 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / aa6d13455b9 |
| Default Java | Private Build-1.8.0_252-8u252-b09-1~18.04-b09 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/1/testReport/ |
| Max. process+thread count | 1619 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/1/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
tested s3 london, unguarded

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  20m 41s |  trunk passed  |
| +1 :green_heart: |  compile  |  17m 31s |  trunk passed  |
| +1 :green_heart: |  checkstyle  |   2m 48s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  5s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  20m 26s |  branch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 27s |  trunk passed  |
| +0 :ok: |  spotbugs  |   1m  7s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  9s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 21s |  the patch passed  |
| +1 :green_heart: |  compile  |  16m 52s |  the patch passed  |
| +1 :green_heart: |  javac  |  16m 52s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 49s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   2m  7s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 24s |  patch has no errors when building and testing our client artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 29s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   3m 28s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   9m 19s |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 28s |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 44s |  The patch does not generate ASF License warnings.  |
|  |   | 123m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2046 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle markdownlint |
| uname | Linux 643ca18e831c 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 40d63e02f04 |
| Default Java | Private Build-1.8.0_252-8u252-b09-1~18.04-b09 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/2/testReport/ |
| Max. process+thread count | 1332 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/2/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
I plan to make the length option a generic fs one so that other filesystems (abfs) can also skip a probe here. a standard option makes it easier to justify hive/orc/parquet switching to this api

File: hadoop-common-project/hadoop-common/src/site/markdown/filesystem/fsdatainputstreambuilder.md [hunk 1]
```diff
@@ -68,6 +68,11 @@ This is relevant with those stores which return version/etag information,
 including the S3A and ABFS connectors -they MAY use this to guarantee that
 the file they opened is exactly the one returned in the listing.
 
+There is no requirement for `status.getPath()` to the resolved path of
```
Comment (mukund-thakur) [inline hadoop-common-project/hadoop-common/src/site/markdown/filesystem/fsdatainputstreambuilder.md:71]: Won't this cause uncertain failures?

**Comment by mukund-thakur** [comment]
Overall this looks really good.

File: hadoop-common-project/hadoop-common/src/site/markdown/filesystem/fsdatainputstreambuilder.md (hunk 1 above)
Comment (steveloughran) [reply hadoop-common-project/hadoop-common/src/site/markdown/filesystem/fsdatainputstreambuilder.md:71]: The reason for this is that viewfs and the hive fs wrappers have their own schemas/paths. If we insist on the paths matching, you woudn't be able to use the API through hive or mounted viewfs, which would be too brittle. Saying "it's just a hint" avoids this issue. Same reason I'm not worrying about type of the status.

s3a FS: use etag and version ID, if known
other status: use the length

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  1s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 3 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   1m  0s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  21m 47s |  trunk passed  |
| +1 :green_heart: |  compile  |  21m  0s |  trunk passed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04  |
| +1 :green_heart: |  compile  |  17m 53s |  trunk passed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09  |
| +1 :green_heart: |  checkstyle  |   2m 53s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m 10s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m  1s |  branch has no errors when building and testing our client artifacts.  |
| -1 :x: |  javadoc  |   0m 35s |  hadoop-common in trunk failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| -1 :x: |  javadoc  |   0m 36s |  hadoop-aws in trunk failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| +1 :green_heart: |  javadoc  |   1m 29s |  trunk passed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09  |
| +0 :ok: |  spotbugs  |   1m 10s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 23s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 24s |  the patch passed  |
| +1 :green_heart: |  compile  |  21m 19s |  the patch passed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04  |
| +1 :green_heart: |  javac  |  21m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |  18m 22s |  the patch passed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09  |
| +1 :green_heart: |  javac  |  18m 22s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   3m  8s |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   2m 15s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedclient  |  15m 34s |  patch has no errors when building and testing our client artifacts.  |
| -1 :x: |  javadoc  |   0m 36s |  hadoop-common in the patch failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| -1 :x: |  javadoc  |   0m 37s |  hadoop-aws in the patch failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| +1 :green_heart: |  javadoc  |   1m 30s |  the patch passed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09  |
| +1 :green_heart: |  findbugs  |   3m 36s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 20s |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 29s |  hadoop-aws in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 47s |  The patch does not generate ASF License warnings.  |
|  |   | 174m 50s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.security.TestRaceWhenRelogin |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2046 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle markdownlint |
| uname | Linux 0bdcb509b029 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 2d12496643b |
| Default Java | Private Build-1.8.0_252-8u252-b09-1~18.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_252-8u252-b09-1~18.04-b09 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/artifact/out/branch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/artifact/out/branch-javadoc-hadoop-tools_hadoop-aws-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/artifact/out/patch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/artifact/out/patch-javadoc-hadoop-tools_hadoop-aws-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/testReport/ |
| Max. process+thread count | 1382 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/4/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=4.0.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +0 :ok: |  markdownlint  |   0m  0s |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 5 new or modified test files.  |
||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  21m 30s |  trunk passed  |
| +1 :green_heart: |  compile  |  21m 20s |  trunk passed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04  |
| +1 :green_heart: |  compile  |  17m 44s |  trunk passed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09  |
| +1 :green_heart: |  checkstyle  |   2m 54s |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  7s |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  20m 43s |  branch has no errors when building and testing our client artifacts.  |
| -1 :x: |  javadoc  |   0m 36s |  hadoop-common in trunk failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| -1 :x: |  javadoc  |   0m 36s |  hadoop-aws in trunk failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| +1 :green_heart: |  javadoc  |   1m 28s |  trunk passed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09  |
| +0 :ok: |  spotbugs  |   1m  8s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 18s |  trunk passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   0m 17s |  hadoop-aws in the patch failed.  |
| -1 :x: |  compile  |  19m  5s |  root in the patch failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| -1 :x: |  javac  |  19m  5s |  root in the patch failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| -1 :x: |  compile  |  16m 36s |  root in the patch failed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09.  |
| -1 :x: |  javac  |  16m 36s |  root in the patch failed with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09.  |
| -0 :warning: |  checkstyle  |   2m 50s |  root: The patch generated 6 new + 152 unchanged - 0 fixed = 158 total (was 152)  |
| -1 :x: |  mvnsite  |   0m 32s |  hadoop-aws in the patch failed.  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 5 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| +1 :green_heart: |  shadedclient  |  15m 41s |  patch has no errors when building and testing our client artifacts.  |
| -1 :x: |  javadoc  |   0m 35s |  hadoop-common in the patch failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| -1 :x: |  javadoc  |   0m 30s |  hadoop-aws in the patch failed with JDK Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.  |
| -1 :x: |  javadoc  |   0m 34s |  hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_252-8u252-b09-1~18.04-b09 with JDK Private Build-1.8.0_252-8u252-b09-1~18.04-b09 generated 1 new + 4 unchanged - 0 fixed = 5 total (was 4)  |
| -1 :x: |  findbugs  |   0m 30s |  hadoop-aws in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   9m 17s |  hadoop-common in the patch passed.  |
| -1 :x: |  unit  |   0m 32s |  hadoop-aws in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 46s |  The patch does not generate ASF License warnings.  |
|  |   | 166m 58s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.fs.contract.rawlocal.TestRawlocalContractOpen |
|   | hadoop.fs.contract.localfs.TestLocalFSContractOpen |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/2046 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle markdownlint |
| uname | Linux 9cc9b3bb8291 4.15.0-91-generic #92-Ubuntu SMP Fri Feb 28 11:09:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / bfcd775381f |
| Default Java | Private Build-1.8.0_252-8u252-b09-1~18.04-b09 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_252-8u252-b09-1~18.04-b09 |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/branch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/branch-javadoc-hadoop-tools_hadoop-aws-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-mvninstall-hadoop-tools_hadoop-aws.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-compile-root-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-compile-root-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-compile-root-jdkPrivateBuild-1.8.0_252-8u252-b09-1~18.04-b09.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-compile-root-jdkPrivateBuild-1.8.0_252-8u252-b09-1~18.04-b09.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/diff-checkstyle-root.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-mvnsite-hadoop-tools_hadoop-aws.txt |
| whitespace | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/whitespace-eol.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-javadoc-hadoop-common-project_hadoop-common-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-javadoc-hadoop-tools_hadoop-aws-jdkUbuntu-11.0.7+10-post-Ubuntu-2ubuntu218.04.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/diff-javadoc-javadoc-hadoop-tools_hadoop-aws-jdkPrivateBuild-1.8.0_252-8u252-b09-1~18.04-b09.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-findbugs-hadoop-tools_hadoop-aws.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-unit-hadoop-common-project_hadoop-common.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/artifact/out/patch-unit-hadoop-tools_hadoop-aws.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/testReport/ |
| Max. process+thread count | 2883 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common hadoop-tools/hadoop-aws U: . |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-2046/5/console |
| versions | git=2.17.1 maven=3.6.0 findbugs=4.0.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
things aren't complaining as the patch chain is confused; a file which was added and then renamed is still present.

I'm going to have to close this PR and create a new one

**Pull request closed without merging**